compiler/rustc_typeck/src/check/writeback.rs

```diff
@@ -475,9 +475,10 @@ impl<'cx, 'tcx> WritebackCx<'cx, 'tcx> {
     }
 
     fn visit_opaque_types(&mut self, span: Span) {
-        for &(opaque_type_key, opaque_defn) in self.fcx.opaque_types.borrow().iter() {
-            let hir_id =
-                self.tcx().hir().local_def_id_to_hir_id(opaque_type_key.def_id.expect_local());
+        for &(opaque_type_key @ OpaqueTypeKey { def_id, substs }, opaque_defn) in
```

Comment:
I'm ok either way, but I personally would prefer to *either* have `opaque_type_key` *or* `OpaqueTypeKey { def_id, substs }`, and not both. If we have both, I have to remember that `def_id` and `substs` are "related" to `opaque_type_key`

Reply:
Done.